lib/libutee/include/tee_internal_api_extensions.h

```diff
@@ -41,4 +42,21 @@ size_t tee_user_mem_check_heap(void);
 /* Hint implementation defines */
 #define TEE_USER_MEM_HINT_NO_FILL_ZERO       0x80000000
 
+/*
+ * Cache maintenance support (TA requires the CACHE_MAINTENANCE property)
+ *
+ * TEE_CacheClean() Write back to memory any dirty data cache lines. The line
+ *                  is marked as not dirty. The valid bit is unchanged.
+ *
+ * TEE_CacheFlush() Purges any valid data cache lines. Any dirty cache lines
+ *                  are first written back to memory, then the cache line is
+ *                  invalidated.
+ *
+ * TEE_CacheInvalidate() Invalidate any valid data cache lines. Any dirty line
+ *                       are not written back to memory.
+ */
+TEE_Result TEE_CacheClean(char *buf, size_t len);
+TEE_Result TEE_CacheFlush(char *buf, size_t len);
+TEE_Result TEE_CacheInvalidate(char *buf, size_t len);
```

Comment:
I would prefer not.


On 5 February 2015 at 16:11, Jérôme Forissier <notifications@github.com>
wrote:

> In lib/libutee/include/tee_internal_api_extensions.h
> <https://github.com/OP-TEE/optee_os/pull/192#discussion_r24168835>:
>
> > +/*
> > + * Cache maintenance support (TA requires the CACHE_MAINTENANCE property)
> > + *
> > + * TEE_CacheClean() Write back to memory any dirty data cache lines. The line
> > + *                  is marked as not dirty. The valid bit is unchanged.
> > + *
> > + * TEE_CacheFlush() Purges any valid data cache lines. Any dirty cache lines
> > + *                  are first written back to memory, then the cache line is
> > + *                  invalidated.
> > + *
> > + * TEE_CacheInvalidate() Invalidate any valid data cache lines. Any dirty line
> > + *                       are not written back to memory.
> > + */
> > +TEE_Result TEE_CacheClean(char *buf, size_t len);
> > +TEE_Result TEE_CacheFlush(char *buf, size_t len);
> > +TEE_Result TEE_CacheInvalidate(char *buf, size_t len);
>
> TEE_CacheOperation(op, buf, len)?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/OP-TEE/optee_os/pull/192/files#r24168835>.
>